### ISSUE TITLE
`ip-pool range add` lacks the necessary options

### ISSUE THREAD

**Issue by ahl** [oxidecomputer/oxide.rs#33]
The `ip-pool range add` subcommand corresponds to the `ip_pool_range_add_v1` operation, the body of which is this type:
```rust
    #[derive(Clone, Debug, Deserialize, Serialize, schemars::JsonSchema)]
    #[serde(untagged)]
    pub enum IpRange {
        V4(Ipv4Range),
        V6(Ipv6Range),
    }
```

This is a little unusual in that most body parameters are `struct`s rather than `enum`s. In this case, both variants of the `enum` are pretty similar:

```rust
    #[doc = "A non-decreasing IPv4 address range, inclusive of both ends.\n\nThe first address must be less than or equal to the last address."]
    #[derive(Clone, Debug, Deserialize, Serialize, schemars::JsonSchema)]
    pub struct Ipv4Range {
        pub first: std::net::Ipv4Addr,
        pub last: std::net::Ipv4Addr,
    }

    #[doc = "A non-decreasing IPv6 address range, inclusive of both ends.\n\nThe first address must be less than or equal to the last address."]
    #[derive(Clone, Debug, Deserialize, Serialize, schemars::JsonSchema)]
    pub struct Ipv6Range {
        pub first: std::net::Ipv6Addr,
        pub last: std::net::Ipv6Addr,
    }
```

Pretty similar, but critically different.

I don't see any great way for progenitor or typify to handle this. For example, I could imaging some sort of enum-builder in situations like this, but it's hard to imagine how one would construct a reasonable error message.

Instead I'm going to create a more general mechanism for
1. modifying the `clap::Command` structures to reflect additional arguments or other changes
2. participating in the construction of the request with these additional arguments (or other changes)

**Comment by david-crespo** [comment]
Same problem with source arg to disk create, I think.

**Comment by ahl** [comment]
Fixed by #57 

Related problem. This case was unique in that the body itself is an `enum` rather than a struct. There are still many missing options.
